packages/tizen_log/lib/tizen_log.dart

```diff
@@ -0,0 +1,166 @@
+  /// indicate problems that the program can tolerate, but should be resolved
+  /// whenever possible.
+  ///
+  /// [file], [func] and [line] parameters are set automatically if debug mode
+  /// is enabled, but they can be explicitly overridden.
+  static void warn(String tag, String message,
+      {String? file, String? func, int? line}) {
+    _log(_LogPriority.warn, tag, message, file: file, func: func, line: line);
+  }
+
+  /// Sends log with ERROR priority and tag. Error log messages are supposed
+  /// to indicate problems that disturb the normal workflow of the application,
+  /// such as functional or performance limitations.
+  ///
+  /// [file], [func] and [line] parameters are set automatically if debug mode
+  /// is enabled, but they can be explicitly overridden.
+  static void error(String tag, String message,
+      {String? file, String? func, int? line}) {
+    _log(_LogPriority.error, tag, message, file: file, func: func, line: line);
+  }
+
+  /// Sends log with FATAL priority and tag. Fatal log messages are supposed
+  /// to indicate problems that entirely block the normal workflow
+  /// of the application.
+  ///
+  /// [file], [func] and [line] parameters are set automatically if debug mode
+  /// is enabled, but they can be explicitly overridden.
+  static void fatal(String tag, String message,
+      {String? file, String? func, int? line}) {
+    _log(_LogPriority.fatal, tag, message, file: file, func: func, line: line);
+  }
+
+  static void _log(_LogPriority priority, String tag, String message,
+      {String? file, String? func, int? line}) {
+    if (isDebugEnabled) {
+      if (file == null || func == null || line == null) {
+        final _StackFrame? frame = _getStackFrameAt(3);
+        if (frame != null) {
+          file ??= frame.file;
+          func ??= frame.function;
+          line ??= frame.line;
+        }
+      }
+      message = '${file ?? "-"}: ${func ?? "-"}(${line ?? "-"}) > $message';
+    }
+    final Pointer<Utf8> tagPtr = tag.toNativeUtf8();
+    final Pointer<Utf8> formatPtr = '%s'.toNativeUtf8();
+    final Pointer<Utf8> messagePtr = message.toNativeUtf8();
+    _dlogPrint(priority.value, tagPtr, formatPtr, messagePtr);
+    malloc.free(tagPtr);
+    malloc.free(formatPtr);
+    malloc.free(messagePtr);
+  }
+
+  static _StackFrame? _getStackFrameAt(int index) {
+    try {
+      final Iterable<RegExpMatch> matches =
+          _stackTraceRegExp.allMatches(StackTrace.current.toString());
+      for (final RegExpMatch match in matches) {
+        final List<String?> groups = match.groups(<int>[1, 2, 3, 4, 5]);
+        if (groups.any((String? group) => group == null) == false) {
+          final int frameIndex = int.parse(groups[0]!);
+          if (frameIndex == index) {
+            return _StackFrame(
+              frameIndex,
+              groups[1]!.trim(),
+              groups[2]!.trim(),
+              int.parse(groups[3]!),
+              int.parse(groups[4]!),
```

Comment:
Can we print only file names and function names as `Tizen.Log` ([TizenFX](https://samsung.github.io/TizenFX/stable/api/Tizen.Log.html)) does?

```
I/ConsoleMessage( 6889): Utils.cs: ParseEngineArgs(30) > Enabled: --enable-dart-profiling
I/ConsoleMessage( 6889): Utils.cs: ParseEngineArgs(30) > Enabled: --enable-checked-mode

W/tizen_log_example( 7304): package:tizen_log_example/main.dart: MyApp.build.<anonymous closure>(48) > warn message
V/tizen_log_example( 7304): package:tizen_log_example/main.dart: MyApp.build.<anonymous closure>(36) > verbose message
```

Reply:
For example only `main.dart` instead of `package:tizen_log_example/main.dart` and only `<anonymous closure>` instead of `MyApp.build.<anonymous closure>`?

Reply:
Yes exactly. That seems to be the standard logging format of all Tizen modules. But we can do it later if the work will take some time.